src/client/components/pages/Courses/RoomSelection.tsx

```diff
@@ -74,14 +79,12 @@ const RoomSelection = (
   return (
     <>
       <RoomSelectionTable
-        roomList={roomList}
+        roomList={roomRequestData ? roomList : []}
```

Comment:
My idea here was to keep anything related to fetching the data in the `RoomSelection` component, and anything specific to showing the data in the `RoomSelectionTable` -- basically making the latter as much of a "dumb box" as possible. Given that, I'd rather keep this logic ("clear the table whenever the query data is cleared") in the smarter parent.

All that being said, having this inlined in the prop like this is kind of a hack, and it would make mor e sense to handle this in the useEffect hook, which I did in 1667fd77

Reply:
Good point